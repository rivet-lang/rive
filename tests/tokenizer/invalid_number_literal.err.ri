0b010101 // ok
0xabcdef // ok
0o1234567 // ok

0b // fail
0o // fail
0x // fail

123eg // fail
123.e // fail
123. // fail
123e // fail
123.abc // fail
123.1.3 // fail

0x_abc // fail
1__00 // fail
0123 // fail
0b01210101 // fail
0xhabc // fail
0o91234 // fail
123_ // fail