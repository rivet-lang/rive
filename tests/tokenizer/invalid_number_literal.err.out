tests/tokenizer/invalid_number_literal.err.ri:5:2: error: number part of this binary number is not provided
      | 
    5 | 0b // fail
      |  ^        
tests/tokenizer/invalid_number_literal.err.ri:6:2: error: number part of this octal number is not provided
      | 
    6 | 0o // fail
      |  ^        
tests/tokenizer/invalid_number_literal.err.ri:7:2: error: number part of this hexadecimal number is not provided
      | 
    7 | 0x // fail
      |  ^        
tests/tokenizer/invalid_number_literal.err.ri:9:5: error: this number has unsuitable digit `g`
      | 
    9 | 123eg // fail
      |     ^        
tests/tokenizer/invalid_number_literal.err.ri:10:5: error: exponent has no digits
      | 
   10 | 123.e // fail
      |     ^        
tests/tokenizer/invalid_number_literal.err.ri:11:4: error: float literals should have a digit after the decimal point
      | 
   11 | 123. // fail
      |    ^        
      = help: use `123.0` instead of `123`
tests/tokenizer/invalid_number_literal.err.ri:12:4: error: exponent has no digits
      | 
   12 | 123e // fail
      |    ^        
tests/tokenizer/invalid_number_literal.err.ri:14:5: error: too many decimal points in number
      | 
   14 | 123.1.3 // fail
      |     ^          
tests/tokenizer/invalid_number_literal.err.ri:16:3: error: separator `_` is only valid between digits in a numeric literal
      | 
   16 | 0x_abc // fail
      |   ^           
tests/tokenizer/invalid_number_literal.err.ri:17:3: error: cannot use `_` consecutively in a numeric literal
      | 
   17 | 1__00 // fail
      |   ^          
tests/tokenizer/invalid_number_literal.err.ri:18:1: error: zeros are not allowed at the beginning of a decimal literal
      | 
   18 | 0123 // fail
      | ^           
      = note: use the prefix `0o` to denote an octal number
tests/tokenizer/invalid_number_literal.err.ri:19:5: error: binary number has unsuitable digit `2`
      | 
   19 | 0b01210101 // fail
      |     ^             
tests/tokenizer/invalid_number_literal.err.ri:20:3: error: hexadecimal number has unsuitable digit `h`
      | 
   20 | 0xhabc // fail
      |   ^           
tests/tokenizer/invalid_number_literal.err.ri:21:3: error: octal number has unsuitable digit `9`
      | 
   21 | 0o91234 // fail
      |   ^            
tests/tokenizer/invalid_number_literal.err.ri:22:4: error: cannot use `_` at the end of a numeric literal
      | 
   22 | 123_ // fail
      |    ^        
error: could not compile source code, aborting due to 15 previous errors